android/CouchbaseLite/src/ce/java/com/couchbase/lite/ReplicatorConfiguration.java

```diff
@@ -15,394 +18,18 @@
-    // seconds; number
-    static final String kC4ReplicatorResetCheckpoint = "reset"; // reset remote checkpoint
-    static final String kC4ReplicatorOptionProgressLevel = "progress";  //< If >=1, notify on every doc; if >=2, on
-    // every attachment (int)
-
-    // Auth dictionary keys:
-    static final String kC4ReplicatorAuthType = "type"; // Auth property; string
-    static final String kCBLReplicatorAuthUserName = "username"; // Auth property; string
-    static final String kCBLReplicatorAuthPassword = "password"; // Auth property; string
-    static final String kC4ReplicatorAuthClientCert = "clientCert"; // Auth property; value platform-dependent
-
-    // auth.type values:
-    static final String kC4AuthTypeBasic = "Basic"; // HTTP Basic (the default)
-    static final String kC4AuthTypeSession = "Session"; // SG session cookie
-    static final String kC4AuthTypeOpenIDConnect = "OpenID Connect";
-    static final String kC4AuthTypeFacebook = "Facebook";
-    static final String kC4AuthTypeClientCert = "Client Cert";
-
-    /**
-     * Replicator type
-     * PUSH_AND_PULL: Bidirectional; both push and pull
-     * PUSH: Pushing changes to the target
-     * PULL: Pulling changes from the target
-     */
-    public enum ReplicatorType {
-        PUSH_AND_PULL,
-        PUSH,
-        PULL
-    }
-
-    //---------------------------------------------
-    // member variables
-    //---------------------------------------------
-
-    private ReplicatorType replicatorType;
-    private boolean readonly;
-    private Database database;
-    private Endpoint target;
-    private boolean continuous;
-    private Authenticator authenticator;
-    private Map<String, String> headers;
-    private byte[] pinnedServerCertificate;
-    private List<String> channels;
-    private List<String> documentIDs;
-    private ReplicationFilter pushFilter;
-    private ReplicationFilter pullFilter;
-
-    //---------------------------------------------
-    // Constructors
-    //---------------------------------------------
-
-    public ReplicatorConfiguration(@NonNull ReplicatorConfiguration config) {
-        if (config == null) { throw new IllegalArgumentException("config cannot be null."); }
-
-        this.readonly = false;
-        this.database = config.database;
-        this.target = config.target;
-        this.replicatorType = config.replicatorType;
-        this.continuous = config.continuous;
-        this.authenticator = config.authenticator;
-        this.pinnedServerCertificate = config.pinnedServerCertificate;
-        this.headers = config.headers;
-        this.channels = config.channels;
-        this.documentIDs = config.documentIDs;
-        this.pullFilter = config.pullFilter;
-        this.pushFilter = config.pushFilter;
+public final class ReplicatorConfiguration extends AbstractReplicatorConfiguration {
+    public ReplicatorConfiguration(@NonNull AbstractReplicatorConfiguration config) {
```

Comment:
The signature should be `public ReplicatorConfiguration(@NonNull ReplicatorConfiguration config)` instead.

Reply:
Why?  This change does not affect the API?